make verilog with "WithNBigCores(32)" fails

<!-- THIS REPOSITORY'S ISSUES ARE RESERVED FOR FEATURE REQUESTS AND BUG REPORTS. -->
<!-- For support questions related to Rocket Chip, please use [Stack Overflow [rocket-chip]](https://stackoverflow.com/questions/tagged/rocket-chip). -->
<!-- For support questions related to Chisel, please use [Stack Overflow [chisel]](https://stackoverflow.com/questions/tagged/chisel). -->
<!-- Please select the item best describing the issue in each category and delete the other items. -->

<!-- choose all that apply -->
**Type of issue**: bug report

<!-- choose one -->
**Impact**: no functional change

<!-- choose one -->
**Development Phase**: request

**Other information**
 <!-- include detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. Stack Overflow, gitter, etc -->
```
Interrupt map (64 harts 3 interrupts):
  [1, 3] => dut

[error] (run-main-0) java.lang.reflect.InvocationTargetException
[error] java.lang.reflect.InvocationTargetException
[error]         at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[error]         at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[error]         at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[error]         at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[error]         at freechips.rocketchip.util.HasGeneratorUtilities.$anonfun$elaborate$1(GeneratorUtils.scala:55)
[error]         at chisel3.core.Module$.do_apply(Module.scala:52)
[error]         at chisel3.Driver$.$anonfun$elaborate$1(Driver.scala:95)
[error]         at chisel3.internal.Builder$.$anonfun$build$2(Builder.scala:343)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
[error]         at chisel3.internal.Builder$.$anonfun$build$1(Builder.scala:341)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
[error]         at chisel3.internal.Builder$.build(Builder.scala:341)
[error]         at chisel3.Driver$.elaborate(Driver.scala:95)
[error]         at freechips.rocketchip.util.HasGeneratorUtilities.elaborate(GeneratorUtils.scala:60)
[error] Caused by: java.lang.IllegalArgumentException: requirement failed: UInt literal -1 is negative
[error]         at scala.Predef$.require(Predef.scala:277)
[error]         at chisel3.internal.firrtl.ULit.<init>(IR.scala:94)
[error]         at chisel3.core.UIntFactory.Lit(Bits.scala:937)
[error]         at chisel3.core.UIntFactory.Lit$(Bits.scala:936)
[error]         at chisel3.core.UInt$.Lit(Bits.scala:953)
[error]         at chisel3.core.package$fromBigIntToLiteral.U(package.scala:36)
[error]         at freechips.rocketchip.devices.debug.TLDebugModuleOuter$$anon$6.$anonfun$new$29(Debug.scala:408)
[error]         at chisel3.core.WhenContext.<init>(When.scala:75)
[error]         at chisel3.core.when$.apply(When.scala:32)
[error]         at freechips.rocketchip.devices.debug.TLDebugModuleOuter$$anon$6.$anonfun$new$27(Debug.scala:407)
[error]         at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:156)
[error]         at freechips.rocketchip.devices.debug.TLDebugModuleOuter$$anon$6.<init>(Debug.scala:397)
[error]         at freechips.rocketchip.devices.debug.TLDebugModuleOuter.module$lzycompute(Debug.scala:289)
[error]         at freechips.rocketchip.devices.debug.TLDebugModuleOuter.module(Debug.scala:289)
[error] Nonzero exit code: 1
[error] (Compile / runMain) Nonzero exit code: 1
[error] Total time: 54 s, completed Oct 31, 2019 2:00:14 PM
```

**If the current behavior is a bug, please provide the steps to reproduce the problem:**
Create a custom Rocket Chip config with `WithNBigCores(32)` and run "make verilog CONFIG=yourconfigname". The makefile will fail and show the error message mentioned above.

Running the same config with only 31 cores works fine.


**What is the current behavior?**
Creating the verilog for a 32 core rocket-chip fails.

**What is the expected behavior?**
Creating the verilog for a 32 core rocket-chip works.

**Please tell us about your environment:**
- version: 86cf7607ed7929a78a21182765aadf97aa30b6cd (my working tree, only thing I changed is the ADMPCIE9h7Config from 16 to 32 cores).

<!--    (examples)
    - version: `git commit hash`
    - OS: `CentOS Linux release 7.6.1810 (Core), Kernel: 3.10.0-862.6.3.el7.x86_64`
-->
**What is the use case for changing the behavior?**
I can synthesize a 32 core rocket-chip. I was not able to find any information about the max number of cores, so I'd expect that to work. Since anything larger than 32 isn't supported by Linux I wouldn't expect anything to work above 32.

--- a/src/main/scala/devices/debug/Debug.scala
+++ b/src/main/scala/devices/debug/Debug.scala
@@ -410,8 +410,8 @@ class TLDebugModuleOuter(device: Device)(implicit p: Parameters) extends LazyMod
       val numHAMASKSlices = ((nComponents - 1)/haWindowSize)+1
       HAWINDOWRdData.maskdata := 0.U     // default, overridden below
       for (ii <- 0 until numHAMASKSlices) {
-        val sliceMask = if (nComponents > ((ii*haWindowSize) + haWindowSize-1)) 0xFFFFFFFF  // All harts in this slice exist
-                        else (1<<(nComponents - (ii*haWindowSize))) - 1         // Partial last slice
+        val sliceMask = if (nComponents > ((ii*haWindowSize) + haWindowSize-1)) (BigInt(1) << haWindowSize) - 1  // All harts in this slice exist
+                        else (BigInt(1)<<(nComponents - (ii*haWindowSize))) - 1         // Partial last slice
         val HAMASKRst = Wire(init = (new HAWINDOWFields().fromBits(0.U)))
         val HAMASKNxt = Wire(init = (new HAWINDOWFields().fromBits(0.U)))
         val HAMASKReg = Wire(init = Vec(AsyncResetReg(updateData = HAMASKNxt.asUInt,

